portal-kernel/src/com/liferay/portal/kernel/security/auth/PrincipalException.java

```diff
@@ -172,7 +172,8 @@ public MustHavePermission(
 			super(
 				String.format(
 					"User %s must have %s permission for %s %s", userId,
-					StringUtil.merge(actionIds, ","), resourceName, resourceId),
+					StringUtil.merge(actionIds, ","), resourceName,
+					(resourceId == 0) ? "" : 0),
```

Comment:
@jorgediaz-lr  Do you really mean ```(resourceId == 0) ? "" : 0)``` here? Or you actually want ```(resourceId == 0) ? "" : resourceId)```

Reply:
Hi @shuyangzhou 

It was a typo, I wanted to put `(resourceId == 0) ? "" : resourceId)` 
(to not write anything in case of  resourceId  == 0)